addons/binding/org.openhab.binding.linuxinput/src/main/java/org/openhab/binding/linuxinput/internal/evdev4j/jnr/EvdevLibrary.java

```diff
@@ -0,0 +1,253 @@
+        public Unsigned16 type = new Unsigned16();
+        public Unsigned16 code = new Unsigned16();
+        public Signed32 value = new Signed32();
+
+        public InputEvent(Runtime runtime) {
+            super(runtime);
+        }
+    }
+
+    final class Handle extends Struct {
+        public Handle(Runtime runtime) {
+            super(runtime);
+        }
+    }
+
+    int new_from_fd(int fd, @Out PointerByReference handle);
+
+    void free(@In Handle handle);
+
+    int grab(@In Handle handle, int grab);
+
+    int next_event(@In Handle handle, int flags, @Out InputEvent event);
+
+    String event_type_get_name(int type);
+
+    String event_code_get_name(int type, int code);
+
+    String event_value_get_name(int type, int code, int value);
+
+    boolean has_event_type(@In Handle handle, int type);
+
+    int enable_event_type(@In Handle handle, int type);
+
+    int disable_event_type(@In Handle handle, int type);
+
+    boolean has_event_code(@In Handle handle, int type, int code);
+
+    int enable_event_code(@In Handle handle, int type, int code);
+
+    int disable_event_code(@In Handle handle, int type, int code);
+
+    String get_name(@In Handle handle);
+
+    void set_name(@In Handle handle, String name);
+
+    String get_phys(@In Handle handle);
+
+    String get_uniq(@In Handle handle);
+
+    int get_id_product(@In Handle handle);
+
+    int get_id_vendor(@In Handle handle);
+
+    int get_id_bustype(@In Handle handle);
+
+    int get_id_version(@In Handle handle);
+
+    int get_driver_version(@In Handle handle);
+
+    @SuppressWarnings("unused")
+    class ReadFlag {
+        private ReadFlag() {
+        }
+
+        public static final int SYNC = 1;
+        public static final int NORMAL = 2;
+        public static final int FORCE_SYNC = 4;
+        public static final int BLOCKING = 8;
+    }
+
+    @SuppressWarnings("unused")
+    class KeyEventValue {
+        private KeyEventValue() {
+        }
+
+        public static final int UP = 0;
+        public static final int DOWN = 1;
+        public static final int REPEAT = 2;
+    }
+
+    @SuppressWarnings("unused")
+    enum Type implements Constant {
+        SYN(0x00),
+        KEY(0x01),
+        REL(0x02),
+        ABS(0x03),
+        MSC(0x04),
+        SW(0x05),
+        LED(0x11),
+        SND(0x12),
+        REP(0x14),
+        FF(0x15),
+        PWR(0x16),
+        FF_STATUS(0x17),
+        MAX(0x1f),
+        CNT(0x20);
+
+        private final int i;
+
+        Type(int i) {
+            this.i = i;
+        }
+
+        public static Optional<Type> fromInt(int i) {
+            return constantFromInt(Type.values(), i);
+        }
+
+        @Override
+        public int intValue() {
+            return i;
+        }
+
+        @Override
+        public long longValue() {
+            return i;
+        }
+
+        @Override
+        public boolean defined() {
+            return true;
+        }
+    }
+
+    @SuppressWarnings("unused")
```

Comment:
What is / why unused here?

Reply:
This concrete enum is actually used.
From the other enums not all values are used, but were copied from the header file for consistency.